frontend/appflowy_flutter/lib/plugins/document/presentation/editor_plugins/heading/heading_toolbar_item.dart

```diff
@@ -0,0 +1,185 @@
+import 'package:appflowy/generated/flowy_svgs.g.dart';
+import 'package:appflowy/generated/locale_keys.g.dart';
+import 'package:appflowy_editor/appflowy_editor.dart';
+import 'package:appflowy_popover/appflowy_popover.dart';
+import 'package:collection/collection.dart';
+import 'package:easy_localization/easy_localization.dart';
+import 'package:flowy_infra_ui/flowy_infra_ui.dart';
+import 'package:flutter/material.dart';
+
+final _headingData = [
+  (FlowySvgs.h1_s, LocaleKeys.editor_heading1.tr()),
+  (FlowySvgs.h2_s, LocaleKeys.editor_heading2.tr()),
+  (FlowySvgs.h3_s, LocaleKeys.editor_heading3.tr()),
+];
+
+final headingsToolbarItem = ToolbarItem(
+  id: 'editor.headings',
+  group: 1,
+  isActive: onlyShowInTextType,
+  builder: (context, editorState, highlightColor, _) {
+    final selection = editorState.selection!;
+    final node = editorState.getNodeAtPath(selection.start.path)!;
+    final delta = (node.delta ?? Delta()).toJson();
+    int level = node.attributes[HeadingBlockKeys.level] ?? 1;
+    final isHighlight =
+        node.type == HeadingBlockKeys.type && (level >= 1 && level <= 3);
+    // only supports the level 1 - 3 in the toolbar, ignore the other levels
+    level = level.clamp(1, 3);
```

Comment:
I guess we can still show if the heading is of another level, we just only allow selecting h1-3 in the toolbar.